lib/elixir/lib/kernel/parallel_compiler.ex

```diff
@@ -319,14 +327,31 @@ defmodule Kernel.ParallelCompiler do
     end
   end
 
+  defp cycle_timing(result, %{profile: :time} = state) do
+    %{cycle_start: cycle_start, module_counter: module_counter} = state
+    num_modules = count_modules(result)
+    now = System.monotonic_time(:millisecond)
+    time = now - cycle_start
+    IO.puts("Finished compilation cycle of #{num_modules - module_counter} modules in #{time}ms")
```

Comment:
Write to `:stderr` and add a prefix? Maybe:

```suggestion
    IO.puts(:stderr, "[profile] Finished compilation cycle of #{num_modules - module_counter} modules in #{time}ms")
```